src/tests/functional/plugin/cpu/shared_tests_instances/subgraph_tests/simple_if.cpp

```diff
@@ -7,51 +7,122 @@
 using namespace SubgraphTestsDefinitions;
 
 namespace {
-    std::vector<std::vector<std::vector<size_t>>> inputShapes = {
-            {{5, 7}, {5, 7}},
-            {{30, 20, 10}, {30, 20, 10}}
-    };
-
-    std::vector<InferenceEngine::Precision> netPrecisions = {InferenceEngine::Precision::FP32,
-                                                             InferenceEngine::Precision::I8,
-    };
-
-    std::vector<bool> conditions = {true, false};
-
-    INSTANTIATE_TEST_SUITE_P(smoke_If, SimpleIfTest,
-                            ::testing::Combine(
-                                    ::testing::ValuesIn(inputShapes),
-                                    ::testing::ValuesIn(netPrecisions),
-                                    ::testing::ValuesIn(conditions),
-                                    ::testing::Values(CommonTestUtils::DEVICE_CPU)),
-                            SimpleIfTest::getTestCaseName);
-
-    TEST_P(SimpleIfTest, CompareWithRefs) {
-        Run();
-    };
-
-    INSTANTIATE_TEST_SUITE_P(smoke_If, SimpleIf2OutTest,
-                            ::testing::Combine(
-                                    ::testing::ValuesIn(inputShapes),
-                                    ::testing::ValuesIn(netPrecisions),
-                                    ::testing::ValuesIn(conditions),
-                                    ::testing::Values(CommonTestUtils::DEVICE_CPU)),
-                            SimpleIf2OutTest::getTestCaseName);
-
-    TEST_P(SimpleIf2OutTest, CompareWithRefs) {
-        Run();
-    };
-
-    INSTANTIATE_TEST_SUITE_P(smoke_If, SimpleIfNotConstConditionTest,
-                            ::testing::Combine(
-                                    ::testing::ValuesIn(inputShapes),
-                                    ::testing::ValuesIn(netPrecisions),
-                                    ::testing::ValuesIn(conditions),
-                                    ::testing::Values(CommonTestUtils::DEVICE_CPU)),
-                            SimpleIfNotConstConditionTest::getTestCaseName);
-
-    TEST_P(SimpleIfNotConstConditionTest, CompareWithRefs) {
-        Run();
-    };
+std::vector<std::vector<ov::test::InputShape>> inputShapes = {
+        {
+            {{}, {{5, 7}}},
```

Comment:
Does Loop support zero dims? It worth testing I think. Example: https://github.com/mandrono/openvino/blob/bf45216260bfe789952d854d57a2dcb8129b4869/src/tests/functional/plugin/cpu/single_layer_tests/nonzero.cpp#L154.
cc'ed @mandrono

Reply:
it's necessary to override `bool isExecutable() const override { return true; }` to `if` work correct with zero dims

Reply:
I took some changes from [#9338](https://github.com/openvinotoolkit/openvino/pull/9338) to cover it as well. Done, thanks